massdriver-application-azure-app-service/main.tf

```diff
@@ -0,0 +1,172 @@
+module "application" {
+  source                  = "github.com/massdriver-cloud/terraform-modules//massdriver-application?ref=51e732e"
+  name                    = var.name
+  service                 = "function"
+  application_identity_id = azurerm_linux_web_app.main.identity[0].principal_id
```

Comment:
Is this the system-assigned identity that "comes with" the Azure app below?

Reply:
That's correct

Reply:
I know this is something you're working on w/ @chrisghill so these are just impressions, don't let me get in your way. :)

Feels like we're doing this for this _one use case_. In the "future" when managed identities are what we use, we'd create that in the mdxc provider and pass it to this function yeah?

What if someone passes in an identity that's not associated with the function. It feels a bit divergent from GCP and AWS to bubble this up outside the provider.

Reply:
Seems like Azure K8s will be using service principals for the foreseeable future. Either way, what the module is now doing is recognizing if an identity already exists (aka, a system assigned identity), and if not, it'll create the service principal for it.

I'm told going forward, MDXC will continue to provide a way for kubernetes and other apps that need service principals to get ones created with the permissions they need. 

We'll also be able to use MDXC to assign RBAC permissions to resources with system assigned identities to provide required permissions to access other resources like storage logs, etc.

Reply:
Sounds good. I'll approve this since I'm out and my feedback has been taken into account. 💪

Reply:
For what it's worth... it may have landed for k8s September 1! I'm digesting them, but docs here. 

https://docs.microsoft.com/en-us/azure/aks/use-managed-identity

Reply:
We are actually looking at this: https://azure.github.io/azure-workload-identity/docs/

We don't want to assign permissions to the entire cluster, we want to assign permissions to individual pods. @chrisghill can explain this a lot better than I can though.